libs/dal/src/repositories/job/job.repository.ts

```diff
@@ -0,0 +1,48 @@
+import { BaseRepository } from '../base-repository';
+import { JobEntity, JobStatus } from './job.entity';
+import { Job } from './job.schema';
+
+export class JobRepository extends BaseRepository<JobEntity> {
+  constructor() {
+    super(Job, JobEntity);
+  }
+
+  public async storeJobs(jobs: JobEntity[]): Promise<JobEntity> {
+    jobs = jobs
+      .map((job: JobEntity) => {
+        job._id = JobRepository.createObjectId();
+
+        return job;
+      })
+      .map((job: JobEntity, index: number, list: JobEntity[]) => {
+        if (index === 0) {
+          return job;
+        }
+        job._parentId = list[index - 1]._id;
+
+        return job;
+      });
```

Comment:
Do you think merging those two maps will work for you? 

```suggestion
    jobs = jobs.map((job: JobEntity) => {
      job._id = JobRepository.createObjectId();

      if (index === 0) {
        return job;
      }
      job._parentId = list[index - 1]._id;

      return job;
    });
```